src/prefect/filesystems.py

```diff
@@ -108,10 +108,13 @@ async def put_directory(self, local_path: str = None, to_path: str = None) -> No
         if local_path is None:
             local_path = Path(".").absolute()
 
-        if sys.version_info < (3, 8):
-            shutil.copytree(from_path, local_path)
+        if local_path == to_path:
+            print("Skipping upload")
```

Comment:
`print` statements are generally not best practice within application code -- I propose that for now we put a `pass` here, and in the future we can wire up some loggers to these objects.  The API only accepts logs attached to flow runs so we'll have to do logging configuration to make that happen which I don't think is worth it just yet
```suggestion
            pass
```